workflows/download_device_design_data_as_landxml/download_device_design_data_as_landxml.py

```diff
@@ -0,0 +1,97 @@
+from file_download import *
+from file_list import *
+from file_features import *
+from metadata_list import *
+
+session = requests.Session()
+
+def particular_from_design_type(a_design_type):
+    if(a_design_type == "Lines"):
+        return "LN3"
+    elif(a_design_type == "Planes"):
+        return "PL3"
+    elif(a_design_type == "Points"):
+        return "PT3"
+    elif(a_design_type == "Roads"):
+        return "RD3"
+    else:
+        return "Invalid"
+
+def main():
+    arg_parser = argparse.ArgumentParser(description="Download device design files as landxml")
+
+    # script parameters:
+    arg_parser = add_arguments_logging(arg_parser, logging.INFO)
+
+    # server parameters:
+    arg_parser = add_arguments_environment(arg_parser)
+    arg_parser = add_arguments_auth(arg_parser)
+
+    arg_parser = add_arguments_pagination(arg_parser)
+
+    # request parameters:
+    arg_parser = add_arguments_site(arg_parser)
+
+    arg_parser.set_defaults()
+    args = arg_parser.parse_args()
+    logging.basicConfig(format=args.log_format, level=args.log_level)
+    # << Arguments
+
+    server = ServerConfig(a_environment=args.env, a_data_center=args.dc)
+
+    logging.info("Running {0} for server={1} dc={2} site={3}".format(os.path.basename(os.path.realpath(__file__)), server.to_url(), args.dc, args.site_id))
+
+    headers = headers_from_jwt_or_oauth(a_jwt=args.jwt, a_client_id=args.oauth_id, a_client_secret=args.oauth_secret, a_scope=args.oauth_scope, a_server_config=server)
+
+    page_traits = MetadataPaginationTraits(a_page_size="500", a_start="")
+    device_design_object_list = query_metadata_by_domain_view(a_server_config=server, a_site_id=args.site_id, a_domain="sitelink", a_view="v_sl_deviceDesignObject_by_deviceURN", a_headers=headers, a_params=page_traits.params())["items"]
+
+    logging.info ("Found {} device design objects".format(len(device_design_object_list)))
+    for fi in device_design_object_list:
+        logging.debug(json.dumps(fi, sort_keys=True, indent=4))
+
+    for fi in device_design_object_list:
+        current_dir = os.path.dirname(os.path.realpath(__file__))
+        output_dir = os.path.join(current_dir, args.site_id[0:12], fi["value"]["deviceURN"].replace(':', "-"))
+        original_dir = os.path.join(output_dir, "original")
+        converted_dir = os.path.join(output_dir, "converted")
+        rdm_file_name = fi["value"]["name"]
+
+        os.makedirs(original_dir, exist_ok=True)
+        os.makedirs(converted_dir, exist_ok=True)
+
+        particular = particular_from_design_type(fi["value"]["designType"])
+        base = "{0}/designfile/v1/sites/{1}/design_files/{2}?design_type={3}".format(server.to_url(), args.site_id, fi["value"]["doFileUUID"], fi["value"]["designType"])
+        url_original = ["{}&particular={}".format(base, particular), particular.lower(), original_dir]
+        url_landxml  = ["{}&particular=LANDXML".format(base, fi["value"]["designType"]), "xml", converted_dir]
```

Comment:
format string "{}&particular=LANDXML" only contain one input (design type is not needed here)

Reply:
I get a 400 when doing this.